File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/ComponentScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.tracker.domain.Component;
import com.esofthead.mycollab.module.tracker.domain.criteria.ComponentSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class ComponentScreenData {
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<Component> {

		public Add(Component component) {
			super(component);
		}
	}

	public static class Edit extends ScreenData<Component> {

		public Edit(Component component) {
			super(component);
		}
	}

	public static class Search extends ScreenData<ComponentSearchCriteria> {

		public Search(ComponentSearchCriteria criteria) {
			super(criteria);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/ComponentScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import com.esofthead.mycollab.module.tracker.domain.Component
import com.esofthead.mycollab.module.tracker.domain.criteria.ComponentSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object ComponentScreenData {

  class Read(params:Integer) extends ScreenData[Integer](params){}

  class Add(params:Component) extends ScreenData[Component](params) {}

  class Edit(params:Component) extends ScreenData[Component](params) {}

  class Search(params:ComponentSearchCriteria) extends ScreenData[ComponentSearchCriteria](params) {}
}
